pkg/app/pipectl/cmd/appconfig/migratefromdeployconfig.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2022 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package appconfig
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/pipe-cd/pipecd/pkg/app/server/service/apiservice"
+	"github.com/pipe-cd/pipecd/pkg/cli"
+)
+
+type migrateFromDeployConfig struct {
+	root *command
+
+	repoRootPath string
+	repoID       string
+	envName      string
+	stdout       io.Writer
+}
+
+func newListCommand(root *command) *cobra.Command {
+	m := &migrateFromDeployConfig{
+		root:   root,
+		stdout: os.Stdout,
+	}
+	cmd := &cobra.Command{
+		Use:   "migrate-from-deploy-config",
+		Short: "Migrate local deployment configuration files to application configuration files. Basically this finds all deployment configuration files to add name field and env label to them.",
+		RunE:  cli.WithContext(m.run),
+	}
+
+	cmd.Flags().StringVar(&m.repoRootPath, "repo-root-path", m.repoRootPath, "The absolute path to the root directory of the Git repository.")
+	cmd.Flags().StringVar(&m.repoID, "repo-id", m.repoID, "The repository ID that is being registered in Piped config.")
+	cmd.Flags().StringVar(&m.envName, "env-name", m.envName, "The environment name.")
+
+	cmd.MarkFlagRequired("repo-root-path")
+	cmd.MarkFlagRequired("repo-id")
+	cmd.MarkFlagRequired("env-name")
```

Comment:
I wonder if `env-name` really needs to be required. For those who want to convert as fast as possible, it's better that they can convert in bulk.

We need the env name at L96, but we can pre-fetch all env and determine the env id from it, so I'd say it's fine. What do you find it?

Reply:
I think that is fine because there are not so many environments. And they can update their dev environment firstly to do some checks before updating other ones.

Reply:
>they can update their dev environment firstly to do some checks before updating other ones.

I feel the same way!
I have no strong opinion but what I want to say is, is it really needed to be required? Just optional enough?